.github/workflows/rust.yml

```diff
@@ -0,0 +1,32 @@
+name: Rust
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install Dependencies and Rustup
+      run: |
+        echo "Installing dependencies and Rustup..."
+        sudo apt-get update
+        sudo apt-get install -y libglib2.0-dev libcairo2-dev libgraphene-1.0-dev libgtk-4-dev libadwaita-1-dev liblua5.4-dev
+        echo "Dependencies and Rustup installed."
```

Comment:
Where do you install rustup ?

Reply:
That was a typo. It used to install rustup but github does that automatically. I'll fix that when I get home.

Reply:
Ah okay